src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -251,9 +251,21 @@ public override int GetHashCode()
         }
 
         /// <summary>
-        /// Returns a <see cref="String"/> with the name of the type and the number of elements
+        /// For <see cref="ReadOnlySpan{Char}"/>, returns a new instance of string that represents the characters pointed to by the span.
+        /// Otherwise, returns a <see cref="String"/> with the name of the type and the number of elements.
         /// </summary>
-        public override string ToString() => string.Format("System.ReadOnlySpan<{0}>[{1}]", typeof(T).Name, _length);
+        public override string ToString()
+        {
+            if (typeof(T) == typeof(char))
+            {
+                unsafe
+                {
+                    fixed (char* src = &Unsafe.As<T, char>(ref _pointer.Value))
+                        return new string(src, 0, _length);
```

Comment:
I assume you're not just using the `string(ReadOnlySpan<char>)` ctor here because this code works on netstandard as well?

Reply:
No, that isn't the reason. This code is netcoreapp specific. Should I just construct a ReadOnlySpan and pass it in? Passing `this` won't work.

Reply:
Once you've gotten a `char*` I don't think there's any benefit to going through the span ctor.

Reply:
> Passing this won't work.

What is the error when you pass this to the constructor? I would think that it should work.

Reply:
> shared\System\ReadOnlySpan.cs(261,35): error CS1503: Argument 1: cannot convert from 'System.ReadOnlySpan<T>' to 'char*
' [D:\GitHub\Fork\coreclr\src\mscorlib\System.Private.CoreLib.csproj]

Reply:
Is the issue that `this` is a `ReadOnlySpan<T>`, but it needs to be a `ReadOnlySpan<char>` in order to pass it to string's constructor?

Reply:
Yes.

Reply:
Ok. I see.